transaction-exclusion-api/app/src/integrationTest/kotlin/net/consensys/linea/transactionexclusion/TransactionExclusionAppTest.kt

```diff
@@ -0,0 +1,291 @@
+        dataQueryableWindowSinceRejectedTimestamp = Duration.parse("P7D")
+      )
+    )
+    app.start().get()
+
+    requestSpecification = RequestSpecBuilder()
+      .setBaseUri("http://localhost:${app.apiBindedPort}/")
+      .build()
+  }
+
+  @AfterEach
+  fun afterEach() {
+    app.stop().get()
+  }
+
+  private fun makeRequestJsonResponse(request: String): String {
+    return RestAssured.given()
+      .spec(requestSpecification)
+      .accept(ContentType.JSON)
+      .body(request)
+      .`when`()
+      .post("/")
+      .then()
+      .statusCode(200)
+      .contentType("application/json")
+      .extract()
+      .asString()
+  }
+
+  private fun saveFirstRejectedTransaction() {
+    val saveTxJonRequest = """{
+      "jsonrpc": "2.0",
+      "id": 123,
+      "method": "linea_saveRejectedTransactionV1",
+      "params": {
+        "txRejectionStage": "P2P",
+        "timestamp": "${Clock.System.now().trimToMillisecondPrecision()}",
+        "transactionRLP": "0x02f8388204d2648203e88203e88203e8941195cf65f83b3a5768f3c496d3a05ad6412c64b38203e88c666d93e9cc5f73748162cea9c0017b8201c8",
+        "reasonMessage": "Transaction line count for module ADD=402 is above the limit 70",
+        "overflows": [
+          { "module": "ADD", "count": 402, "limit": 70 },
+          { "module": "MUL", "count": 587, "limit": 401 }
+        ]
+      }
+    }
+    """.trimIndent()
+
+    // Check the save response and ensure the rejected txn was saved
+    assertThatJson(makeRequestJsonResponse(saveTxJonRequest))
+      .isEqualTo(
+        """{
+          "jsonrpc": "2.0",
+          "id": 123,
+          "result": {"status":"SAVED","txHash":"0x526e56101cf39c1e717cef9cedf6fdddb42684711abda35bae51136dbb350ad7"}
+        }"""
+      )
+  }
+
+  @Test
+  fun `Should save the rejected tx from P2P and then from SEQUENCER with same txHash but different reason message`() {
+    // Save the first rejected tx from P2P without rejected block number
+    saveFirstRejectedTransaction()
+
+    // Save the rejected tx from SEQUENCER with rejected block number and different
+    // rejected reason and a more recent rejected timestamp
+    val rejectionTimeStamp = Clock.System.now()
+      .trimToMillisecondPrecision()
+      .toString()
+
+    val saveTxJonRequest = """{
+      "jsonrpc": "2.0",
+      "id": 124,
+      "method": "linea_saveRejectedTransactionV1",
+      "params": [{
+        "txRejectionStage": "SEQUENCER",
+        "timestamp": "$rejectionTimeStamp",
+        "transactionRLP": "0x02f8388204d2648203e88203e88203e8941195cf65f83b3a5768f3c496d3a05ad6412c64b38203e88c666d93e9cc5f73748162cea9c0017b8201c8",
+        "blockNumber": "10000",
+        "reasonMessage": "Transaction line count for module ADD=402 is above the limit 70 (from Sequencer)",
+        "overflows": [
+          { "module": "ADD", "count": 402, "limit": 70 },
+          { "module": "MUL", "count": 587, "limit": 401 }
+        ]
+      }]
+    }
+    """.trimIndent()
```

Comment:
this is a bit of repeated code. Why not make a `saveRejectedTx` which gets timestamp, txRejectionStage, etc fields as argument with defaults? this saveRejectedTx would only save and return the response (no assertions)

Reply:
imo, the way it is now is better for readability, it shows the request and response in json plain text in each test case, I'd prefer to keep it this way